test_app/migrations/0003_create_system_user.py

```diff
@@ -12,16 +10,16 @@ def create_system_user(apps, schema_editor):
     The user is self-referential in its created_by and modified_by fields.
     It is inactive, only used for attributing internal changes to the system.
     """
-    User = get_user_model()
+    User = apps.get_model('test_app', 'User')
```

Comment:
This works for test_app, but it might be nice to still write it generally. To do that, I expect

```python
User = apps.get_model(settings.AUTH_USER_MODEL)
```

Would work the same, and still wouldn't encounter the problems you're fixing here about importing a model from current apps in migrations.

Reply:
I'm not sure this is a good idea. The proxy model used in this migration relies on the concrete model defined in the previous migration. The implementation here also relies on the specific set of fields defined in that model.

Reply:
I'm lost at "proxy model". I don't know where any proxy models exist in test_app or DAB.

Reply:
Okay, maybe I used an inaccurate term for this, because in Django proxy models is a different very specific thing.

The Django migrations engine tracks state of previous migrations and `apps.get_model` returns you a virtual model based on this state. In this specific case the virtual `User` model generated from the model created in the `0001_initial` migration.

This migration relies on implementation details of that model and therefore requests it explicitly.
If the user model ever changes, it will require either introducing new migrations or rebuilding the entire migration tree.

Depending on `settings.AUTH_USER_MODELS` and using `swappable_dependency()` is useful for the library authors, when the specific model defined by the end-user is unknown in advance.

Reply:
Ok, I agree with everything you're doing. But I also still think `settings.AUTH_USER_MODEL` is valid, because it will return the prior state of the model. It's the same thing as `"test_app.User"`. It's just a string, and just a minor nit.